clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetchRequestManager.java

```diff
@@ -82,6 +82,7 @@ public PollResult poll(long currentTimeMs) {
      */
     @Override
     public PollResult pollOnClose() {
+        // TODO: move the logic to poll to handle signal close
```

Comment:
I added a method signalClose() to the interface.  I wonder if we should keep letting the network thread poll the network client as usual, until we actually invoke close.  This means, close will do very little but check if there are any pending requests.

Reply:
@kirktrue

Reply:
Yes, using the normal poll loop sounds like a good idea. We should still probably  `sendUnsentRequests` once when the timeout has passed.